var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -94,21 +95,24 @@ def install(self, spec, prefix):
             config_args.append('--enable-parallel4')
 
         CPPFLAGS.append("-I%s/include" % spec['hdf5'].prefix)
-        LDFLAGS.append("-L%s/lib"     % spec['hdf5'].prefix)
+        LDFLAGS.append("-L%s/lib" % spec['hdf5'].prefix)
+        for lib in os.listdir(join_path(spec['hdf5'].prefix, 'lib')):
+            if lib.endswith('.so'):
```

Comment:
@adamjstewart #1682 should help with things like this by : 
- encapsulating common transformations of library paths in a class (`LibraryList`)
- providing a function `find_libraries` that helps in creating instances of `LibraryList`

The PR is ready as far as I am concerned, and I am just waiting for further reviews or a merge.